tests/sample_qc/test_de_novo.py

```diff
@@ -0,0 +1,230 @@
+            expected_hemi_x,
+            expected_hemi_y,
+        ], (
+            f"Failed for locus={locus}, is_xx={is_xx}. Expected"
+            f" {[expected_diploid, expected_hemi_x, expected_hemi_y]}, got {result}"
+        )
+
+    @pytest.mark.parametrize(
+        "locus_key, proband_gt, father_gt, mother_gt, is_xx, expected",
+        [
+            ("autosomal", (0, 1), (0, 0), (0, 0), False, True),
+            ("autosomal", (1, 1), (0, 0), (0, 0), False, False),
+            ("x_nonpar", (1, 1), None, (0, 0), False, True),
+            ("x_nonpar", (1, 1), (0, 0), None, False, None),
+            ("y_nonpar", (1, 1), None, None, False, None),
+        ],
+    )
+    def test_call_de_novo(
+        self, locus_key, proband_gt, father_gt, mother_gt, is_xx, expected
+    ) -> None:
+        """
+        Test de novo mutation detection with different loci and parental genotypes.
+
+        :param locus_key: Locus key.
+        :param proband_gt: Proband genotype.
+        :param father_gt: Father genotype.
+        :param mother_gt: Mother genotype.
+        :param is_xx: Whether the individual is XX.
+        :param expected: Expected de novo mutation status.
+        :return: None.
+        """
+        locus_expr = self.loci[locus_key]
+        proband_expr = hl.struct(
+            GT=hl.call(*proband_gt) if proband_gt else hl.missing(hl.tcall)
+        )
+        father_expr = hl.struct(
+            GT=hl.call(*father_gt) if father_gt else hl.missing(hl.tcall)
+        )
+        mother_expr = hl.struct(
+            GT=hl.call(*mother_gt) if mother_gt else hl.missing(hl.tcall)
+        )
+        is_xx_expr = hl.literal(is_xx)
+
+        expr = call_de_novo(
+            locus_expr, proband_expr, father_expr, mother_expr, is_xx_expr
+        )
+        result = hl.eval(expr)
+
+        assert (
+            result == expected
+        ), f"Mismatch in {locus_key}: Expected {expected}, got {result}"
+
+    def test_calculate_de_novo_post_prob(self):
+        """Test posterior probability computation for de novo mutations."""
+        expr = calculate_de_novo_post_prob(
+            self.proband_pl,
+            self.father_pl,
+            self.mother_pl,
+            self.locus_expr,
+            self.is_xx_expr,
+            self.freq_prior_expr,
+        )
+        result = hl.eval(expr)
+        assert 0 <= result <= 1  # Posterior probability should be within valid range
+
+    def test_get_de_novo_expr(self):
+        """Test the de novo expression struct output."""
+        expr = get_de_novo_expr(
+            self.locus_expr,
+            self.alleles_expr,
+            self.proband_expr,
+            self.father_expr,
+            self.mother_expr,
+            self.is_xx_expr,
+            self.freq_prior_expr,
+        )
+        result = hl.eval(expr)
+
+        assert "p_de_novo" in result
+        assert "confidence" in result
+        assert 0 <= result.p_de_novo <= 1  # Probability must be valid
```

Comment:
similar question to above: why not check the actual value produced by your test values?